src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java

```diff
@@ -596,30 +597,30 @@ static String getSPMetadataFilePath() {
      * @return the http response.
      */
     @SuppressWarnings("unused")
-    public HttpResponse doMetadata(StaplerRequest request, StaplerResponse response) {
+    public HttpResponse doMetadata(StaplerRequest2 request, StaplerResponse2 response) {
         return new SamlSPMetadataWrapper(getSamlPluginConfig(), request, response).get();
     }
 
     /**
-     * @see SecurityRealm#getPostLogOutUrl
+     * @see SecurityRealm#getPostLogOutUrl2
      * Note: It does not call the logout service on SAML because the library does not implement it on this version,
      * it will be done when we upgrade the library.
```

Comment:
@kuisathaverat Now that the library has been upgraded, this task can be completed. From [the blog post](https://www.pac4j.org/blog/what_s_new_in_pac4j_v6.html):

> The `processLogout` method has been added, based on the `LogoutProcessor` component.

Reply:
I have limited time, I do not plan to do it soon